Snabble/Core/Payment/PaymentMethodDetails.swift

```diff
@@ -604,35 +687,60 @@ public enum PaymentMethodDetails {
         storage.save(details)
     }
 
-    static func save(_ detail: PaymentMethodDetail, at index: Int? = nil) {
-        storage.save(detail, at: index)
+    static func save(_ detail: PaymentMethodDetail) {
+        storage.save(detail)
     }
 
-    static func remove(at index: Int) {
-        storage.remove(at: index)
+    static func remove(_ detail: PaymentMethodDetail) {
+        storage.remove(detail)
     }
 
-    public static func startup(_ firstStart: Bool) {
+    //
+    /// initialize the storage for payment methods
+    /// - Parameter firstStart: when `true` (on the first start of the app), remove all stored payment methods
+    /// - Returns: true when any obsolete payment methods had to be removed
+    ///
+    /// This method silently removes any payment methods that have expired, most notably credit cards past
+    /// their expiration date.
+    public static func startup(_ firstStart: Bool) -> Bool {
```

Comment:
Ist es mit dem `firstStart` notwendig? Ich bin mir nicht sicher, ob ich es richtig verstanden habe, aber könnten wir nicht alle gespeicherten Kreditkarten auslesen und dann alle auf Basis von `version` `1` (CreditCardData) entfernen?

Reply:
Firststart ist nötig für den Fall dass du nach dem Löschen der App sie wieder neu installierst, dabei sollen die alten Records aus der Keychain gelöscht werden. Der check mit v1-CC-Daten ist davon aber unabhängig.

Reply:
Oh gut zu wissen, auf die Anforderung wäre ich niemals gekommen. Schade, für den Nutzer wäre es doch eigentlich ganz hilfreich wenn die Daten wieder verwendet werden, weißt du noch warum es notwendig ist die vorhanden Informationen zu löschen? Reines Interesse.

Reply:
Reine Psychologie: weil wir nicht den Eindruck erwecken wollen dass wir diese Infos doch irgendwo gespeichert haben. Abgesehen davon gehe ich davon aus dass Apple diesen Bug auch irgendwann mal fixen wird und dann die Info eh weg ist.

Reply:
Letzteres glaube ich nicht, da es die Keychain ist. Wenn ich auf macOS eine App lösche sind die Daten, die durch sie in die Keychain geschrieben sind auch weiterhin vorhanden.